src/main/java/teammates/storage/entity/CourseStudent.java

```diff
@@ -0,0 +1,272 @@
+package teammates.storage.entity;
+
+import java.security.SecureRandom;
+import java.util.Date;
+
+import javax.jdo.annotations.Extension;
+import javax.jdo.annotations.NotPersistent;
+import javax.jdo.annotations.PersistenceCapable;
+import javax.jdo.annotations.Persistent;
+import javax.jdo.annotations.PrimaryKey;
+import javax.jdo.listener.StoreCallback;
+
+import teammates.common.util.Assumption;
+import teammates.common.util.StringHelper;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * An association class that represents the association Account -->
+ * [enrolled in] --> Course.
+ */
+@PersistenceCapable
+public class CourseStudent implements StoreCallback {
+    /**
+     * Setting this to true prevents changes to the lastUpdate time stamp.
+     * Set to true when using scripts to update entities when you want to
+     * preserve the lastUpdate time stamp.
+     **/
+    @NotPersistent
+    public transient boolean keepUpdateTimestamp;
+
+    /**
+     * Format: email%courseId e.g., adam@gmail.com%cs1101
+     */
+    @PrimaryKey
+    @Persistent
+    private String id;
+    
+    @Persistent
+    private Date createdAt;
+    
+    @Persistent
+    private Date updatedAt;
+    
+    
+    /**
+     * Copied from old student class
+     * Null if using new registration key instead.
+     */
+    @Persistent
+    private transient String oldRegistrationKey;
```

Comment:
If this is going to be filled during migration, we might as well migrate everything to `registrationKey` since old -> new is just a matter of encryption, right?